src/wallet/wallet.cpp

```diff
@@ -2358,6 +2358,13 @@ bool CWallet::SelectCoins(const std::vector<COutput>& vAvailableCoins, const CAm
             ++it;
     }
 
+    unsigned int limit_ancestor_count;
+    unsigned int limit_descendant_count;
+    chain().getPackageLimits(limit_ancestor_count, limit_descendant_count);
+    size_t max_ancestors = (size_t)std::max<int64_t>(1, limit_ancestor_count);
```

Comment:
I don't know. I did some digging and the type was introduced [here](https://github.com/bitcoin/bitcoin/pull/12634/files#diff-b2bb174788c7409b671c46ccc86034bdR2588) and prior to that a `uint64` was first introduced [here](https://github.com/bitcoin/bitcoin/pull/9262/files#diff-b2bb174788c7409b671c46ccc86034bdR2020) in this context. I can only speculate that since this is a value from user input it was supposed to be as forgiving as possible if a ridiculously high number is provided, but even that does not make too much sense to me.

Reply:
This may have been a way to safeguard against a user defined value that was negative, but the switch to `getPackageLimits` removed that safeguard. Not sure if that was wise or not, but that's a separate question.